Added multiple titles

LegacyValken opened the pull request:
Added thanos snap titles.
Added bounty collection titles.
Added button finding titles.

LegacyValken committed:
Added three titles.

Added thanos snap titles.
Added bounty collection titles.
Added button finding titles.

swampservers commented:
tyvm, ill get back to this early next week (have irl shit to do) but feel free to add more to this pr

LegacyValken committed:
Title update.

Added multiple new titles.
Added new server table TitleFuncs.
Added new server function TitlesPlayerDeath.
Added new server hook TitlesPlayerDeath.
Fixed thanos snap table not correctly being ordered.
Removed unnecessary logic in gauntlet snap.

LegacyValken commented:
I added some more titles and did some logic for deaths, it's been a while so let me know if I did the functions within tables wrong at all.

LegacyValken committed:
Fix typo in bounty kill title.

LegacyValken committed:
Title changes.

Added dodgeball titles "Underdog" and "Strongarm".
Changed snap kill requirements.
Changed kleiner kill rewards.
Changed fist kill rewards.
Changed dodgeball rewards.

LegacyValken committed:
Fix missing zero on fist kills.

LegacyValken committed:
Titles requested by Purapoint.

Added snowball hitting title requested by Purapoint.
Added energy drinking title requested by Purapoint.
Renamed TitleFuncs to TitleFunctions for better clarity.

Noztik commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
+    { 100, "Button Master", 0}
+}, "Find and press %s buttons.", "s_magicbutton")
+
+AddTitle("snowballhit", {
```
probably a bad idea since snowballs are only available during december

LegacyValken replied:
> probably a bad idea since snowballs are only available during december

So is giving gifts though?

swampservers replied:
No, the mystery box is a gift. Probably can make the snowball appear in the mysterybox to address this...

Metroid55 replied:
Aren't you getting snowballs when you stand on the vents on the roofs?

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -213,3 +276,25 @@ AddTitle("", {
+if SERVER then
+    local TitleFunctions =
+    {
+        ["weapon_gauntlet"] = function(atk) atk:AddStat("snaps") end,
```
for string keys do `keyname=value` not `["keyname"]=value`

LegacyValken replied:
Think I got it.

LegacyValken replied:
Throwing this into gluaint, it's not happy about the key being a string and thinks that `keyname=value` has an unexpected `=`. Is this just gluaint being weird?

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -213,3 +276,25 @@ AddTitle("", {
+        ["weapon_fists"] = function(atk) atk:AddStat("fistkill") end
+    }
+
+    local function TitlesPlayerDeath(vic,inf,atk)
```
`hook.Add("PlayerDeath", "TitlesPlayerDeath", function(vic,inf,atk)    ...    end)`

LegacyValken replied:
While I'll edit it so that the logic is in the hook.Add instead of a separate function, I will ask why the need for the change. I assume less code in the script?

swampservers replied:
It is cleaner. The function logic is part of the same statement as the hook.Add. You don't need to name the function.

LegacyValken replied:
Ah. This is more than fair.

LegacyValken committed:
Use strings not bracketed strings.

LegacyValken committed:
Put logic in hook.Add, not separate function.

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -213,3 +276,24 @@ AddTitle("", {
+        if atk:IsPlayer() then
+            if TitleFuncions[inf:GetClass()] then TitleFunctions[inf:GetClass()](atk,vic) end
+            if vic:IsPlayer() && not vic:IsAfk() && vic:LastHitGroup() == 1 then atk:AddStat("headshotkill") end
+            if vic:GetModel() == "models/player/kleiner.mdl" then atk:AddStat("kleinerkiller") end
```
kleinerkill for consistency. once you set this its a hassle to change it

LegacyValken replied:
Done.

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
+    {10, "Bounty Hunter", 0},
+    {100, "The Cleaner", 0},
+    {1000, "Hitman", 0}
+}, "Collect %s bounties.", "s_bounties")
```
s_bountykill

use a partnerset so the bounties are unique

LegacyValken replied:
That is a good point, I didn't consider it for unique kills to make it more difficult to acquire. Done.

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
+}, "Get %s kills with the fists.", "s_fistkill")
+
+AddTitle("dodgeballkill", {
+    {25, "FatKid", 2500},
```
should use titles like "Jock"

LegacyValken replied:
I can't think of any besides Jock and Bully which I have slotted in.

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
+}, "Kill %s non-afk players with the throatneck slitter.", "s_knifekill")
+
+AddTitle("theaterkill", {
+    {100, "Peacekeeper", 0}
```
Protector, Guardian, Homeland Security, The Law (last 2 would be leaderboard stats, not fully implemented)

LegacyValken replied:
Sounds good, let me know if I need to tweak it any further now that I've changed them to this.

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
+    {1000, "Hitman", 0}
+}, "Collect %s bounties.", "s_bounties")
+
+AddTitle("headshotkill", {
```
Can you please just remove all the rewards for now. Remove the third item in the tables that have rewards and set the first string ("headshotkill" here) to ""

LegacyValken replied:
Done. Let me know if I did something wrong with it.

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
 }, "Be subjected to one of Dr. Isaac Kleiner's teleportation experiments", "s_kleinertp")
 
+AddTitle("buttonsearch", {
+    { 1, "Button Rookie", 0},
```
Curious, Sharp eye, Seeker, Hunter

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
+}, "Hit %s active players with a snowball.", "s_snowballhit")
+
+AddTitle("drinkenergy", {
+    {100, "Boomer"},
```
this should be like something where you have to drink monster and then stand around until the effect wears off to increment the stat

LegacyValken replied:
Yeah I'm fine with doing that.

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
+AddTitle("kleinerkiller", {
+    {100, "Kleiner Killer", 5000},
+    {1000, "Anti Kleiner", 10000},
+    {10000, "Kleiner Exterminator", 25000}
```
Last title should just be Exterminator

swampservers commented on lua/swamp/misc/sh_titles.lua:
```diff
@@ -160,6 +160,69 @@ AddTitle("kleinertp", {
+AddTitle("knifekill", {
+    {25, "Edgelord", 0},
+    {100, "Stab Enthusiast", 0},
+    {1000, "American Psycho", 0}
```
Good titles

LegacyValken committed:
Requested changed.

Added multiple theater protection titles.
Changed bounty kill stat from AddStat to SetPartnerStat.
Changed fist kill titles to better reflect it's theme.
Changed snap bounty title function and reward function for better consistency.
Changed button search titles to recommended titles.
Changed title function and reward function for kleiner kills to kleinerkill for consistency.
Changed reward function for bounty kill to s_bountykill for consistency.
Move energy drink AddStat from drink to timer.
Removed headshot kill rewards.
Removed third headshot kill title.

LegacyValken committed:
Dev work.

Revert fist kill titles.
Changed dodgeball kill titles.

LegacyValken committed:
Fix bad formatting for table.
